netlib/tcp.py

```diff
@@ -728,7 +731,18 @@ def convert_to_ssl(self, sni=None, alpn_protos=None, **sslctx_kwargs):
 
     def connect(self):
         try:
-            connection = socket.socket(self.address.family, socket.SOCK_STREAM)
+            if not self.connection:
+                connection = socket.socket(self.address.family, socket.SOCK_STREAM)
+            else:
+                connection = self.connection
```

Comment:
It enables users to make use of the server_conn stub, to patch the socket. For example:

`def serverconnect(server_conn):
    rooter = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    rooter.connect('/tmp/rooter')

    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sendfd(rooter, server_socket)
    fd, _ = recvfd(rooter)
    
    server_conn.connection = socket.fromfd(fd, socket.AF_INET, socket.SOCK_STREAM)`